src/service/history-impl.js

```diff
@@ -374,6 +403,9 @@ export class HistoryBindingNatural_ {
     // On pop, stack is not allowed to go prior to the starting point.
     stackIndex = Math.max(stackIndex, this.startIndex_);
     return this.whenReady_(() => {
+      // Popping history forget the last navigated hash since we can't really
+      // know what hash the browser is going to go to.
+      this.lastNavigatedHash_ = null;
```

Comment:
Yes. But we don't really know what's that going to be (Browser history API doesn't expose that for privacy reasons). However, we set the hash in the `popstate` event once we already been back in history.